eng/common/pipelines/templates/steps/verify-links.yml

```diff
@@ -1,19 +1,21 @@
 parameters:
   Directory: 'not-specified'
   IgnoreLinksFile: "$(Build.SourcesDirectory)/eng/ignore-links.txt"
-  
+  WorkingDirectory: $(System.DefaultWorkingDirectory)
+  ScriptDirectory: eng/common/scripts
 steps:
-- task: PowerShell@2
-  displayName: Link verification check
-  inputs:
-    pwsh: true
-    workingDirectory: $(Build.SourcesDirectory)/${{ parameters.Directory }}
-    filePath: eng/common/scripts/Verify-Links.ps1 
-    arguments: >
-      -urls $(dir -r -i *.md) 
-      -rootUrl "file://$(Build.SourcesDirectory)/${{ parameters.Directory }}" 
-      -recursive: $false 
-      -ignoreLinksFile ${{ parameters.IgnoreLinksFile }}
-      -branchReplaceRegex "^($env:SYSTEM_PULLREQUEST_SOURCEREPOSITORYURI.*/(?:blob|tree)/)master(/.*)$"
-      -branchReplacementName $env:SYSTEM_PULLREQUEST_SOURCECOMMITID
-      -devOpsLogging: $true
+  - task: PowerShell@2
+    displayName: Link verification check
+    inputs:
+      pwsh: true
+      workingDirectory: "${{ parameters.WorkingDirectory }}/${{ parameters.Directory }}"
+      filePath: ${{ parameters.ScriptDirectory }}/Verify-Links.ps1
+      arguments: >
+        -urls $(dir -r -i *.md) 
+        -rootUrl "file://$(Build.SourcesDirectory)/${{ parameters.Directory }}" 
+        -recursive: $false 
+        -ignoreLinksFile ${{ parameters.IgnoreLinksFile }}
+        -branchReplaceRegex "^($env:SYSTEM_PULLREQUEST_SOURCEREPOSITORYURI.*/(?:blob|tree)/)master(/.*)$"
+        -branchReplacementName $env:SYSTEM_PULLREQUEST_SOURCECOMMITID
+        -devOpsLogging: $true
+        -checkLinkGuidance: $true
```

Comment:
We need to make all these parameters options to the yml template so we can control them as needed.

Reply:
Let do this for all the parameters: urls, recursive, branchreplaceregex, branchreplacementname, checklinkguidance